pkg/lib/verifier/rule_builder_util.go

```diff
@@ -17,112 +18,174 @@ import (
+//return nil if error occurs
+func getPcrEventLogEqualsExcludingRules(pcrs []types.PcrIndex, pcrLogData *types.PCRS, flavor *hvs.Flavor, marker common.FlavorPart) ([]rules.Rule, error) {
 	var results []rules.Rule
 
-	// iterate over the banks, collecting the values for each supplied index
-	// and create PcrEventLogEquals rules (when present).
-	for bank, pcrMap := range flavor.Pcrs {
-		for _, index := range pcrs {
-			if expectedPcrEx, ok := pcrMap[index.String()]; ok {
-
-				expectedEventLogEntry := types.EventLogEntry{
-					PcrIndex:  index,
-					PcrBank:   types.SHAAlgorithm(bank),
-					EventLogs: expectedPcrEx.Event,
-				}
-
-				rule, err := rules.NewPcrEventLogEquals(&expectedEventLogEntry, flavor.Meta.ID, marker)
-				if err != nil {
-					return nil, errors.Wrapf(err, "An error occurred creating a PcrEventLogEqualsExcluding rule for bank '%s', index '%s'", bank, index)
+	if pcrLogData != nil {
+		expectedPcrEventLogEntry := types.EventLogEntryFC{
+			Pcr: types.PCR{
+				Index: pcrLogData.PCR.Index,
+				Bank:  pcrLogData.PCR.Bank,
+			},
+			TpmEvent: pcrLogData.EventlogEqual.Events,
+		}
+		rule, err := rules.NewPcrEventLogEqualsExcluding(nil, &expectedPcrEventLogEntry, nil, pcrLogData.EventlogEqual.ExcludeTags, uuid.Nil, marker)
+		if err != nil {
+			return nil, errors.Wrapf(err, "An error occurred creating a PcrEventLogEqualsExcluding rule for bank '%s', index '%d'", pcrLogData.PCR.Bank, pcrLogData.PCR.Index)
+		}
+		results = append(results, rule)
+	} else {
+
+		// iterate over the banks, collecting the values for each supplied index
+		// and create PcrEventLogEqualsExcluding rules (when present).
+		for bank, pcrMap := range flavor.Pcrs {
+			for _, index := range pcrs {
+				if expectedPcrEx, ok := pcrMap[index.String()]; ok {
+
+					expectedEventLogEntry := types.EventLogEntry{
+						PcrIndex:  index,
+						PcrBank:   types.SHAAlgorithm(bank),
+						EventLogs: expectedPcrEx.Event,
+					}
+					expectedPcr, _ := rules.FlavorPcr2ManifestPcr(&expectedPcrEx, types.SHAAlgorithm(bank), index)
+					rule, err := rules.NewPcrEventLogEqualsExcluding(&expectedEventLogEntry, nil, expectedPcr, nil, flavor.Meta.ID, marker)
+					if err != nil {
+						return nil, errors.Wrapf(err, "An error occurred creating a PcrEventLogEqualsExcluding rule for bank '%s', index '%s'", bank, index)
+					}
+
+					results = append(results, rule)
 				}
-
-				results = append(results, rule)
 			}
 		}
 	}
 
 	return results, nil
 }
 
-func getPcrEventLogIntegrityRules(pcrs []types.PcrIndex, flavor *hvs.Flavor, marker common.FlavorPart) ([]rules.Rule, error) {
+//getPcrEventLogIntegrityRules method will create PcrEventLogIntegrityRule and return the rule
+//return nil if error occurs
+func getPcrEventLogIntegrityRules(pcrs []types.PcrIndex, flavor *hvs.Flavor, pcrLogData *types.PCRS, marker common.FlavorPart) ([]rules.Rule, error) {
 
 	var results []rules.Rule
```

Comment:
Using variable names as 'rules' instead of 'results' is showing error because we are using package name called 'rules'. Hence it is conflicting. Can we use some other name like 'pcrRules'?